staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -857,7 +854,7 @@ func OwningAttachments(obj interface{}) (owners []string, err error) {
 	owners = make([]string, 0, 1)
 	for _, oref := range meta.GetOwnerReferences() {
 		if oref.Kind == "NetworkAttachment" && oref.Controller != nil && *oref.Controller {
-			owners = append(owners, oref.Name)
+			owners = append(owners, meta.GetNamespace()+"/"+oref.Name)
```

Comment:
This avoids unnecessary work: when an attachment  X is processed this indexer is used to retrieve and consider all the IPLocks owned by X. Without this change the retrieval is more liberal than it should: locks owned by attachments with the same name as X but different namespace are also returned. The retrieval is still more liberal than it could be because locks owned by a deleted attachment with the same namespaced name as X could be returned. We can use the owner UID as the index to be as stringent as we can. Notice that  before this change
locks owned by attachments with the same name as X but different namespace were correctly detected as non-belonging to X thanks to a check on the owner's UID but erroneously classified and logged as time slippers.

Reply:
OK